### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by Turbo87**
@stefanpenner this should probably be reviewed on a commit-by-commit basis. the last two commits in particular are things I'm not 100% sure about, but in case I understood the code correctly they should work fine.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -33,7 +33,6 @@ describe('capture-exit', function() {
   describe('.captureExit', function() {
     afterEach(function() {
       // always restore, in case we have bugs in our code while developing
-      process.exit = originalExit;
```
Comment (stefanpenner) [inline test.js:35]: I think we should keep this one. I think we should restore using abstractions not implemented in this library. As if this library breaks, our tests will break for unrelated reasons

File: index.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ module.exports.captureExit = function() {
       })
       .catch(function(error) {
         console.error(error);
-        exit.apply(process, 1);
+        exit.call(process, 1);
```
Comment (stefanpenner) [inline index.js:38]: apparently then, this code-path isn't tested

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: test.js (hunk 1 above)
Comment (Turbo87) [reply test.js:35]: as mentioned in the commit message: the outer `afterEach()` block is doing the same thing already. I don't see a reason to reset this twice.

File: index.js (hunk 2 above)
Comment (Turbo87) [reply index.js:38]: yes, that test appears to be missing

**Comment by Turbo87** [comment]
@stefanpenner do you need me to do any changes here? The test was already missing before the PR so we can add that in a separate PR afterwards. Feel free to just cherry pick the good commits for now if you disagree with parts of this PR.

**Comment by stefanpenner** [comment]
@Turbo87 opened an issue: https://github.com/stefanpenner/capture-exit/issues/3 so we dont forget :)

**Pull request merged**